src/qibosoq/client.py

```diff
@@ -34,7 +34,7 @@ def connect(server_commands: dict, host: str, port: int) -> Tuple[list, list]:
         results = json.loads(received.decode("utf-8"))
         if isinstance(results, str) and "Error" in results:
             raise QibosoqError(results)
-        return results["i"], results["q"]
+        return results["i"], results["q"]  # type: ignore
```

Comment:
here I suppose we could do something like : `results : Union[Dict, str]`

Reply:
But a string can only be accessed with integers, not with other strings. How would it be possible at runtime to still have a string here?

The error most likely survives because you're checking with `and` above, but maybe you'd need more something like:
```py
if isinstance(results, str):
    if "Error" in results:
        raise QibosoqError(results)
    raise ValueError("something")
```

I know it's Pythonic to fail if an operation is not possible. But what is Pythonic actually change a bit when you start using types (it's quite uncontroversial that types, even with inference, are a different paradigm from the run-time-only Python).

Reply:
This indeed works, but then I would simply remove the second if:
```python
if isinstance(results, str):
    raise QibosoqError(results)
```
It still makes enough sense right?